doc/developers/contributing.rst

```diff
@@ -952,6 +952,18 @@ The easiest and recommended way to accomplish this is to
 All logic behind estimator parameters,
 like translating string arguments into functions, should be done in ``fit``.
 
+Parameters can be passed to the ``fit`` method. Generally, these parameters
+should be restricted to variables that have the shape ``n_samples``.
+This allows for all relevant parameters to be sliced cleanly during
+cross-validation. All other parameters should be passed to
+``__init__`` or ``set_params``.
```

Comment:
Parameters that affect behavior of fit does should always go into __init__. The idea with the fit parameters is to accommodate variables like sample_weights, which need to sliced up along with the X and Y during cross-validation. That should be clear in the documentation - I'll make the change.